src/kOS.Safe/Compilation/KS/Compiler.cs

```diff
@@ -575,10 +561,37 @@ private void PreProcessUserFunctionStatement(ParseNode node)
                     AddOpcode(new OpcodeReturn());
                 }
                 userFuncObject.ScopeNode = GetContainingBlockNode(node); // This limits the scope of the function to the instruction_block the DEFINE was in.
-                userFuncObject.IsFunction = true;
+                userFuncObject.IsFunction = !(isLock);;
             }
         }
         
+        
+        /// <summary>
+        /// Build the system trigger to go with a user function (lock)
+        /// such as LOCK STEERING or LOCK THROTTLE
+        /// </summary>
+        /// <param name="func">Represents the lock object, which might not be fully populated yet.</param>
+        private void BuildSystemTrigger(UserFunction func)
+        {
```

Comment:
This logic used to be occurring when preparsing the lock statement, the very first time it encountered this particular system variable being bound.  But now that it's possible to visit things out of order (where the triggers get compiled before the locks do), it's important to get this code ready up front at the start of the compile regardless of what order things happen in.  This is the trigger that sets boundvariable $STEERING and $THROTTLE once per FixedUpdate by executing the function in the lock $steering* and $throttle*.  Without it being in place, flybywire cannot work.